### ISSUE TITLE
New Metrics Aggregator and Visualization in Production

### ISSUE THREAD

**Issue by djbrooke** [IQSS/dataverse-metrics#7]
We want to replace the Miniverse visualizations at dataverse.org/metrics with the new, aggregated visualizations. This will require standing up the metrics aggregator and visualization somewhere and pointing the URL to it instead of Miniverse. 

This does not involve taking down Miniverse, as we should keep Miniverse around for the map on dataverse.org.

**Comment by djbrooke** [comment]
Release: 

https://github.com/IQSS/dataverse-metrics/releases/tag/v0.2.0

**Comment by djbrooke** [comment]
Consider putting this on the Dataverse map server, or minting a SSL cert for this.

**Comment by pdurbin** [comment]
@donsizemore and I chatted about this issue at http://irclog.iq.harvard.edu/dataverse/2019-03-28#i_89412

I asked if UNC would be willing to host an installation of dataverse-metrics for the community and he said they would. Thanks!

@djbrooke gave his blessing and suggested that if UNC wanted to add some branding to the bottom of the page, that's fine. He also gave me edit access to dataverse.org and I can see where I need to update the iframe URL once the new host is ready. Here are some screenshots:

![Screen Shot 2019-03-28 at 4 07 02 PM](https://user-images.githubusercontent.com/21006/55189346-0bf25d80-5174-11e9-9206-41ed344587bb.png)

![Screen Shot 2019-03-28 at 4 07 07 PM](https://user-images.githubusercontent.com/21006/55189345-0b59c700-5174-11e9-930b-b429591f99ec.png)

**Comment by donsizemore** [comment]
@pdurbin certs requested. probably tomorrow.

**Comment by donsizemore** [comment]
[https://dataversemetrics.odum.unc.edu](https://dataversemetrics.odum.unc.edu)

**Comment by pdurbin** [comment]
@donsizemore thanks!

I just updated the iframe URL I mentioned above, changing

`<iframe src="https://services.dataverse.harvard.edu/miniverse/metrics/basic-viz/last12-dataverse-org?iframe=true" width="100%" height="3600px" frameborder="0"></iframe>`

... to

`<iframe src="https://dataversemetrics.odum.unc.edu/dataverse-metrics/" width="100%" height="3600px" frameborder="0"></iframe>`

Here are screenshots of "before" (Harvard Dataverse only) and "after" (12 installations of Dataverse around the world) from https://dataverse.org/metrics

![metrics-before-harvard-dataverse-only](https://user-images.githubusercontent.com/21006/55241002-8ec8f600-5210-11e9-822b-9189a2b68710.png)

![metrics-after-12-installations](https://user-images.githubusercontent.com/21006/55241001-8ec8f600-5210-11e9-96d2-29d4ece2be0d.png)

It looks and behaves a little strange from my Android phone. I'll explain at standup in 20 minutes.

**Comment by pdurbin** [comment]
@djbrooke @dlmurphy and I discussed the "after" screenshot and situation after standup and I'm going to try to force a single column from OpenScholar.

The main concern we have on mobile is that the text disappears for months unless your screen is big enough. It looks better on iPhone than Android. On Android, there's a tiny scroll window.

**Comment by pdurbin** [comment]
Drat. I was trying to change `width="100%"` to  `width="750px"` but now OpenScholar won't even let me save, giving me this error:

"This embed code failed validation. Please check that all urls are with https and from accepted domains"

Maybe we need to whitelist unc.edu? I have no idea why I was able to use a unc.edu address an hour ago but not now.

**Comment by pdurbin** [comment]
Oh, I flipped the metrics page back to miniverse (the "before" above) since I can no longer use unc.edu in an iframe.

**Comment by pdurbin** [comment]
![Screen Shot 2019-03-29 at 11 47 40 AM](https://user-images.githubusercontent.com/21006/55245317-3a764400-5219-11e9-9af3-d5667ceda5d6.png)

**Comment by pdurbin** [comment]
Ok, @djbrooke got unc.edu whitelisted 🎉 and I constrained the width to 750 pixels like this:

`<iframe src="https://dataversemetrics.odum.unc.edu/dataverse-metrics/" width="750px" height="4000px" frameborder="0"></iframe>`

@djbrooke and I also removed the scroll bars by increasing the height. Now we have a single column that looks more reasonable on desktop (still a little weird on Android, in my opinion, a tiny viewport). Here's a screenshot from Firefox on Mac:

![after2-width750](https://user-images.githubusercontent.com/21006/55246941-bfaf2800-521c-11e9-85a2-43222983e701.png)

**Comment by pdurbin** [comment]
Here's how it looks on my Android phone:

![Screenshot_20190329-141302](https://user-images.githubusercontent.com/21006/55253960-1a508000-522d-11e9-9d8e-888cb75ab175.png)

**Comment by TaniaSchlatter** [comment]
@pdurbin no ability to scroll? Is this bringing the metrics in via iframe on an open scholar page?

**Comment by donsizemore** [comment]
@pdurbin I should probably duck as I click the "comment" button, but I have received a request to make all Dataverse instance links at the bottom open to target=_new

**Comment by pdurbin** [comment]
@donsizemore I'm fine with that but please open an issue so it can be discussed.

@TaniaSchlatter you can scroll but it's weird and you have to see it on an Android phone to believe it. Yes, the metrics have always been in an iframe. The only difference is that yesterday the iframe was coming from miniverse and today the iframe is coming from the code in this repo (dataverse-metrics).

**Comment by pdurbin** [comment]
Something's not quite right. I just looked again at the before and after screenshots at https://github.com/IQSS/dataverse-metrics/issues/7#issuecomment-478027524 and the numbers seem off:

- before (Harvard Dataverse only): ~55K datasets
- after (12 installations of Dataverse): ~36K datasets

I'm assuming this will fix itself when the cron job runs within the next 24 hours.

**Comment by pdurbin** [comment]
Huh, https://dataverse.harvard.edu/api/info/metrics/datasets/toMonth only returns ~28K datasets.

I guess this means that miniverse (~55K) and the Metrics API count datasets very differently.

Also @donsizemore mentioned yesterday that because https://data.inra.fr is down right now, they aren't being counted and that had a huge number of datasets (~77K) when I was working on #4. Most of these had a subject of "Not specified", which I commented on at https://github.com/IQSS/dataverse-metrics/issues/4#issuecomment-472464194 . I'm used to seeing higher numbers (~36K + ~77K = ~113K) but we did observe that most of the Data Inra datasets didn't have any files so I have mixed feelings about keeping them on the list. Maybe until they're back up we should take them out of config.json so they don't show up at the bottom of the page.

The other thing I'd like to mention is that @donsizemore noticed that the links to the installations at the bottom should probably open in a new tab or window. I hadn't played with these links while the page is embedded within OpenScholar until just a few minutes ago and you get some strange behavior with the installation embedded within dataverse.org sometimes. There's a similar bug report having to do with clicking installations from the map (also an iframe within OpenScholar) at https://github.com/IQSS/miniverse/issues/64

**Comment by TaniaSchlatter** [comment]
@donsizemore I agree the Dataverse instance links at the bottom should open to target=_new.

**Comment by TaniaSchlatter** [comment]
@pdurbin if I narrow the browser window, I see what you show above in the Android image, and I can scroll. I can get all the information, only in a tiny content area is awkward because the footer is fixed. I think the Open Scholar template would need to be adjusted to improve that. 

The layout is ok on my iPhone. The footer is not fixed, so the content area is fine. However, no chart text shows – no Chart headers or labels on X/Y axis. I see some text on tapping the charts, but not all. 

![Image-1 (1)](https://user-images.githubusercontent.com/15386424/55277437-9a7ef000-52d6-11e9-89ce-aac11f46951c.jpg)

**Comment by djbrooke** [comment]
Thanks @donsizemore ! I'm going to close this issue since the metrics visualization is now up on dataverse.org/metrics.
